app/lib/methods/updateLastOpen.ts

```diff
@@ -0,0 +1,21 @@
+import database from '../database';
+import log from './helpers/log';
+import { TSubscriptionModel } from '../../definitions';
+
+export async function updateLastOpen(rid: string, lastOpen = new Date()): Promise<void> {
+	try {
+		const db = database.active;
+		const subscription = await db.get('subscriptions').find(rid);
+		await db.write(async () => {
+			try {
+				await subscription.update((s: TSubscriptionModel) => {
+					s.lastOpen = lastOpen;
+				});
+			} catch (e) {
```

Comment:
Unless I'm missing something, I'm pretty sure you could remove this whole try/catch block and rely only on the outer try/catch.